src/transform/transform.ts

```diff
@@ -697,7 +697,7 @@ function createResponseEnvelope(codeModel: CodeModel, group: OperationGroup, op:
   // if this is an ARM spec and this is a HEAD method then return a BooleanResponse
   // response envelope instead of http.Response.  only do this if the operation doesn't model any
   // header values (HAB will be enabled for that response envelope).
-  if (op.language.go!.openApiType === 'arm' && op.requests![0].protocol.http!.method === 'head' && headers.size === 0) {
+  if (codeModel.language.go!.headAsBoolean && op.requests![0].protocol.http!.method === 'head' && headers.size === 0) {
```

Comment:
Are there going to be openApiType === 'arm' that don't have headAsBoolean  set in azure-sdk-for-go?

Reply:
In the SDK proper it's highly unlikely as all the ARM config files (should) specify `azure-arm:true`.

Where this mostly gets funky is in the autorest test server.  Since autorest defaults to `arm`, you get ARM semantics by default and a lot of the tests assume this.  However, we should only be enabling these Azure-specific things if `azure-arm` is set to `true` although again, a lot of the test server tests kind of make this assumption (see the ones where I added the `head-as-boolean` flag in this PR).

Also, while I doubt this would happen, there's no reason a data-plane service shouldn't be allowed to specify `head-as-boolean:true` if they wanted to.